core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ambari/AmbariDecommissioner.java

```diff
@@ -650,27 +661,38 @@ private PollingResult waitForRegionServerDecommission(Stack stack, AmbariClient
     private void retireNifiNodes(Stack stack, AmbariClient ambariClient, List<String> hostList, Map<String, HostStatus> hostStatusMap) {
         List<String> hostsRunService = hostList.stream().filter(hn -> hostStatusMap.get(hn).getHostComponentsStatuses().containsKey(NIFI))
                 .collect(Collectors.toList());
-        hostsRunService.stream().forEach(nifiNode -> {
+        hostsRunService.forEach(nifiNode -> {
             LOGGER.info("Retiring NiFi node {}", nifiNode);
             Pair<PollingResult, Exception> retireResult = Pair.of(SUCCESS, null);
             try {
                 int requestId = ambariClient.retire(List.of(nifiNode), "NIFI", NIFI);
                 retireResult = ambariOperationService.waitForOperations(stack, ambariClient,
                         singletonMap("Retiring NiFi node", requestId), RETIRE_NIFI_NODE_AMBARI_PROGRESS_STATE);
             } catch (Exception e) {
-                handleRetireError(nifiNode, e);
+                LOGGER.warn("Retire command failed.", e);
```

Comment:
if `ambariOperationService.waitForOperations` throws an exception, `retireResult` will be SUCCESS based on line 666 and we will not handle failure. I would set default result for `retireResult` to failed or I would give new value to `retireResult` here